rxpager/src/test/java/com/schibstedspain/android/rxpager/PagerTest.java

```diff
@@ -34,131 +36,125 @@
 
+    waitUntilLoaded(() -> pager.getPageObservable().subscribe(testObserver));
     waitUntilLoaded(() -> pager.next());
 
-    verify(getPageMock, times(2)).call(anyString());
-    testSubscriber.assertReceivedOnNext(Arrays.asList(FIRST_PAGE, SECOND_PAGE));
+    verify(getPageMock, times(2)).apply(anyString());
+    testObserver.assertValueSequence(Arrays.asList(FIRST_PAGE, SECOND_PAGE));
   }
 
-  private void waitUntilLoaded(Action0 action) {
-    TestSubscriber<Boolean> loadingSubscriber = new TestSubscriber<>();
+  private void waitUntilLoaded(Action action) throws Exception {
+    TestObserver<Boolean> loadingSubscriber = new TestObserver<>();
     pager.getIsLoadingObservable().take(3).subscribe(loadingSubscriber);
-    action.call();
-    loadingSubscriber.awaitTerminalEvent();
+    action.run();
+    loadingSubscriber.awaitTerminalEvent(2, TimeUnit.SECONDS);
   }
 
   @Test
   public void isLoadingObservableShouldReturnFalseBeforeAskingForOnePage() {
 
-    Boolean isLoadingBeforeSubscribe = pager.getIsLoadingObservable().toBlocking().first();
+    Boolean isLoadingBeforeSubscribe = pager.getIsLoadingObservable().blockingFirst();
 
     assertFalse(isLoadingBeforeSubscribe);
   }
 
   @Test
-  public void isLoadingObservableShouldReturnTrueWhileIsWaitingForAPage() {
-    PublishSubject<TokenPage<String>> getPageSubject = PublishSubject.<TokenPage<String>>create();
-    given(getPageMock.call(anyString())).willReturn(getPageSubject);
+  public void isLoadingObservableShouldReturnTrueWhileIsWaitingForAPage() throws Exception {
+    PublishSubject<TokenPage<String>> getPageSubject = PublishSubject.create();
+    given(getPageMock.apply(anyString())).willReturn(getPageSubject);
     pager.getPageObservable().subscribe();
 
-    Boolean isLoading = pager.getIsLoadingObservable().toBlocking().first();
+    Boolean isLoading = pager.getIsLoadingObservable().blockingFirst();
 
     assertTrue(isLoading);
   }
 
   @Test
-  public void isLoadingObservableShouldReturnFalseAfterReceivingOnePage() {
-    given(getPageMock.call(anyString())).willReturn(Observable.just(FIRST_PAGE));
+  public void isLoadingObservableShouldReturnFalseAfterReceivingOnePage() throws Exception {
+    given(getPageMock.apply(anyString())).willReturn(Observable.just(FIRST_PAGE));
     pager.getPageObservable().take(1).subscribe();
 
-    Boolean isLoading = pager.getIsLoadingObservable().toBlocking().first();
+    Boolean isLoading = pager.getIsLoadingObservable().blockingFirst();
 
     assertFalse(isLoading);
   }
 
   @Test
-  public void isLoadingObservableShouldReturnFalseAfterAFailure() {
+  public void isLoadingObservableShouldReturnFalseAfterAFailure() throws Exception {
     givenGetPageFails();
-    TestSubscriber<TokenPage<String>> testSubscriber = new TestSubscriber<>();
+    TestObserver<TokenPage<String>> testSubscriber = new TestObserver<>();
     pager.getPageObservable().subscribe(testSubscriber);
     testSubscriber.awaitTerminalEvent();
```

Comment:
Please use the `.test()` instead of using a `TestObserver` here, it will be fluent and nicer.

Reply:
done